src/main/java/com/teragrep/cfe_39/consumers/kafka/DatabaseOutput.java

```diff
@@ -254,16 +258,37 @@ public void accept(List<RecordOffset> recordOffsetObjectList) {
             }
 
             byte[] byteArray = recordOffsetObject.getRecord(); // loads the byte[] contained in recordOffsetObject.getRecord() to byteArray.
-            batchBytes = batchBytes + byteArray.length;
+            if (byteArray == null) {
+                if (skipEmptyRFC5424Records) {
+                    if (LOGGER.isDebugEnabled()) {
+                        LOGGER
+                                .debug(
+                                        "Skipping processing an empty non RFC5424 record. Record metadata: {}",
+                                        recordOffsetObject.offsetToJSON()
+                                );
+                    }
+                    continue;
+                }
+                else {
+                    if (LOGGER.isDebugEnabled()) {
+                        LOGGER.debug("Null record metadata: {}", recordOffsetObject.offsetToJSON());
+                    }
+                    syslogFile.delete(); // Clean up
+                    throw new NullPointerException("Record with null content detected during processing.");
```

Comment:
throw custom exception instead of NPE because the case of null is handled by removing the file.